### DISCUSSION TITLE
[PHP]Is there any way to set SkyWalking token authentication?

### DISCUSSION THREAD

**Discussion by huange7** [apache/skywalking#9860] [Q&A] [answered]
I am using the php version of skywalking and found that php.ini does not provide the configuration for **authentication**, is it not supported here?

**Comment by wu-sheng** [comment, marked as answer]
I think it doesn't yet. @jmjoy

**Reply by huange7** [reply]
Thanks for your reply. Is there any other way to solve it?

**Reply by wu-sheng** [reply]
Open source is a thing from people's contribution. If someone does, there is. Otherwise, nothing happens. 
So, always recommend you take action, rather than wait.

I don't know what you mean `other way`. Contributing this new feature is the best one.

**Reply by huange7** [reply]
thanks a lot. ^_^

**Comment by bearxiongzhen** [comment]
Why does java language skywalking have this field?php does not？

**Reply by jmjoy** [reply]
Because nobody has done it yet.

**Comment by heyanlong** [comment]
Authentication is a java Agent to configure. If you need it, you can add a new feature to skywalking-php

**Reply by bearxiongzhen** [reply]
This configuration exists in older versions(SkyAPM-php-sdk](https://github.com/SkyAPM/SkyAPM-php-sdk))：skywalking.oap_authentication

Why is there no current version?

**Reply by heyanlong** [reply]
The new Agent development plan has nothing to do with the old Agent, The origin Agent is only for reference

**Comment by jmjoy** [comment]
Already supported in https://github.com/apache/skywalking-php/releases/tag/v0.3.0
